src/bgg_companion_api.py

```diff
@@ -67,17 +76,8 @@ def get_object_ids(user: str) -> list[str]:
         return id_list
 
 
-def get_random_game(user: str) -> Union[BoardGame, UserIsNoneError]:
-    try:
-        id_list = get_object_ids(user)
-    except UserIsNoneError as exc:
-        return exc
-    id_string = ",".join(id_list)
-    board_games = get_board_games(id_string)
-    if board_games is None:
-        raise BoardGameIsNoneError("A board game was passed that does not exist within BoardGameGeek.")
-    for game in board_games:
-        if game.type != "boardgame":
-            board_games.remove(game)
-    game = random.choice(board_games)
+def get_random_game(user: str, maxplayers=None, exactplayers=None) -> BoardGame:
+    ids_list = get_users_game_ids(user)
+    board_games = get_board_games(tuple(ids_list))
+    game = random.choice(FilterBoardGame().filter_games(board_games, GameFilter(maxplayers=maxplayers, exactplayers=exactplayers)))
```

Comment:
This line feels like there's a lot going on lol.  I understand it but I don't know if it's too complicated.  Or maybe it's just very "Pythonic"

Reply:
I think you could break them out into multiple lines if you think it's too tricky to grok

```python
filtered_board_games = (FilterBoardGame().filter_games(board_games, GameFilter(maxplayers=maxplayers, exactplayers=exactplayers))
return random.choice(filtered_board_games)
```

Reply:
Ahhhhh yeah actually seeing it broken out vs all together in one line, I think it's more readable to break it out.  Thanks!